cordova-common/src/FileUpdater.js

```diff
@@ -0,0 +1,389 @@
+ *     set, then the file will not be copied unless the source is newer than the target.
+ * @param {function} [log] Optional logging callback that takes a string message describing any
+ *     file operations that are performed.
+ * @return {boolean} true if any changes were made, or false if the force flag is not set
+ *     and everything was up to date
+ */
+function updatePath(rootDir, targetPath, sourcePath, force, log) {
+    rootDir = rootDir || "";
+    if (typeof(rootDir) !== "string") {
+        throw new Error("A root directory path is required.");
+    }
+
+    if (!targetPath || typeof(targetPath) !== "string") {
+        throw new Error("A target path is required.");
+    }
+
+    if (sourcePath && typeof(sourcePath) !== "string") {
+        throw new Error("A source path (or null) is required.");
+    }
+
+    log = log || function(message) { };
+
+    var targetFullPath = path.join(rootDir, targetPath);
+    var targetStats = fs.existsSync(targetFullPath) ? fs.statSync(targetFullPath) : null;
+    var sourceStats = null;
+
+    if (sourcePath) {
+        // A non-null source path was specified. It should exist.
+        var sourceFullPath = path.join(rootDir, sourcePath);
+        if (!fs.existsSync(sourceFullPath)) {
+            throw new Error("Source path does not exist: " + sourcePath);
+        }
+
+        sourceStats = fs.statSync(sourceFullPath);
+
+        // Create the target's parent directory if it doesn't exist.
+        var parentDir = path.dirname(targetFullPath);
+        if (!fs.existsSync(parentDir)) {
+            shell.mkdir("-p", parentDir);
+        }
+    }
+
+    return updatePathWithStats(
+        rootDir, targetPath, targetStats, sourcePath, sourceStats, force, log);
+}
+
+/**
+ * Updates files and directories based on a mapping from target paths to source paths. Targets
+ * with null sources in the map are deleted.
+ *
+ * @param {string|null} rootDir Root directory (such as a project) to which target and source
+ *     path parameters are relative, or null if the paths are absolute. The rootDir is omitted
+ *     from any logged paths, to make the logs easier to read.
+ * @param {object} pathMap A dictionary mapping from target paths to source paths.
+ * @param {boolean} force If target and source are both files, and the force flag is not
+ *     set, then the file will not be copied unless the source is newer than the target.
+ * @param {function} [log] Optional logging callback that takes a string message describing any
+ *     file operations that are performed.
+ * @return {boolean} true if any changes were made, or false if the force flag is not set
+ *     and everything was up to date
+ */
+function updatePaths(rootDir, pathMap, force, log) {
+    if (!pathMap || typeof(pathMap) !== "object") {
+        throw new Error("An object mapping from target paths to source paths is required.");
+    }
+
+    var updated = false;
+
+    for (var targetPath in pathMap) {
```

Comment:
Yes, I'm fixing these.

---
In reply to: [60399365](https://github.com/apache/cordova-lib/pull/429#discussion_r60399365) [](ancestors = 60399365)